Serpent/Serpent/Classes/Extensions/URLSession+Codable.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  URLSession+Codable.swift
+//  Serpent
+//
+//  Created by Andrei Hogea on 16/03/2018.
+//  Copyright © 2018 Nodes. All rights reserved.
+//
+
+import Foundation
+
+// Decoded Result
+public enum DResult<Value> {
+    case success(Value)
+    case failure(Error)
+}
+
+public extension URLSession {
+    
+    /**
+     Adds a handler that attempts to parse the result of the request into a **Decodable**
+     
+     - parameter requestCompletion: The URLSession.dataTask completion
+     
+     - returns: The Decoded Result (DResult)
+     */
+    public func decode<Value: Swift.Decodable>(requestCompletion: (Data?, Error?)) -> DResult<Value> {
+        let (data, error) = requestCompletion
+        
+        if let error = error {
+            return .failure(error)
+        }
+        
+        if let data = data {
+            do {
+                let decodedData = try JSONDecoder().decode(Value.self, from: data)
+                return .success(decodedData)
+            } catch let decodeError {
+                return .failure(decodeError)
```

Comment:
In terms of simplicity it is nice to have just a simple one value .failure and .success, however I think it would be nice to get the data back even if the decoding fails (same as comment for Alamofire implementation basically). Otherwise debugging it will be annoying, so the more data we can pass back the better IMO.